pkg/api/server/v1alpha2/results_test.go

```diff
@@ -102,6 +104,69 @@ func TestCreateResult(t *testing.T) {
 	}
 }
 
+func TestUpdateResult(t *testing.T) {
+	srv, err := New(test.NewDB(t))
+	if err != nil {
+		t.Fatalf("failed to create server: %v", err)
+	}
+	ctx := context.Background()
+
+	tt := []struct {
+		name    string
+		create  *pb.Result
+		update  *pb.Result
+		expect  *pb.Result
+		errcode codes.Code
+	}{
+		{
+			name:   "test success",
+			update: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+		},
+		{
+			name:   "test update with empty result",
+			expect: &pb.Result{},
+		},
+	}
+	for idx, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			// create a result for test.
+			r := &pb.Result{Name: fmt.Sprintf("foo/results/bar-%v", idx)}
+			created, err := srv.CreateResult(ctx, &pb.CreateResultRequest{Parent: "foo", Result: r})
+			if err != nil {
+				t.Fatalf("could not create result: %v", err)
+			}
+
+			tc.expect.Name = created.GetName()
+			tc.expect.Id = created.GetId()
+			tc.expect.CreatedTime = created.GetCreatedTime()
+
+			// foward the time to test if the UpdateTime field is properly updated.
+			fakeClock.Advance(time.Second)
+
+			updateRequest := &pb.UpdateResultRequest{Result: tc.update, Name: created.GetName()}
+			updated, err := srv.UpdateResult(ctx, updateRequest)
+			if err != nil {
+				if status.Code(err) == tc.errcode {
+					return
+				}
+				t.Fatalf("UpdateResult()=(%v, %v); want %v", r, err, tc.errcode)
+			}
+
+			tc.expect.UpdatedTime = timestamppb.New(clock.Now())
+
+			got, err := srv.GetResult(ctx, &pb.GetResultRequest{Name: updated.GetName()})
+			if err != nil {
+				t.Fatalf("failed to get result from server: %v", err)
+			}
+
+			if diff := cmp.Diff(tc.expect, got, protocmp.Transform()); diff != "" {
```

Comment:
Good idea checking that the Update is reflected by the Get, but we should also check that the response from Update is what we expect.

Reply:
Great!